lib/Form/Processor/RoleFormProcessor.php

```diff
@@ -38,20 +38,22 @@ public static function getSubscribedEvents()
     public function processDefaultAction(FormActionEvent $event)
     {
         // Don't update anything if we just want to cancel the draft.
-        if ($event->getClickedButton() === 'removeDraft') {
+        // Don't do the default action when saving is specified, as it's the same action and shouldn't be done twice.
+        if (in_array($event->getClickedButton(), ['removeDraft', 'saveRole'])) {
```

Comment:
I'd abstract it to a boolean test, maybe `$this->skipSave($event)`. It would read a bit better, and would avoid the inline comment (that would then be on the method's phpdoc).